src/pybind/mgr/dashboard/controllers/multi_cluster.py

```diff
@@ -375,25 +431,110 @@ def check_token_status(self, clustersTokenMap=None):
         clusters_token_map = json.loads(clustersTokenMap)
         return self.check_token_status_array(clusters_token_map)
 
+    @Endpoint()
+    @ReadPermission
+    def security_config(self):
+        return self._get_security_config()
+
+    def _get_security_config(self):
+        orch_backend = mgr.get_module_option_ex('orchestrator', 'orchestrator')
+        if orch_backend == 'cephadm':
+            cmd = {
+                'prefix': 'orch get-security-config',
+            }
+            ret_status, out, _ = mgr.mon_command(cmd)
+            if ret_status == 0 and out is not None:
+                security_info = json.loads(out)
+                security_enabled = security_info['security_enabled']
+                mgmt_gw_enabled = security_info['mgmt_gw_enabled']
+                return {
+                    'security_enabled': bool(security_enabled),
+                    'mgmt_gw_enabled': bool(mgmt_gw_enabled)
+                }
+        return None
+
     @Endpoint()
     @ReadPermission
     def get_prometheus_api_url(self):
+        security_content = self._get_security_config()
+        mgmt_gw_enabled = security_content['mgmt_gw_enabled']
         prometheus_url = Settings.PROMETHEUS_API_HOST
+
         if prometheus_url is not None:
-            # check if is url is already in IP format
-            pattern = r'^(?:https?|http):\/\/(?:\d{1,3}\.){3}\d{1,3}:\d+$'
-            valid_ip_url = bool(re.match(pattern, prometheus_url))
-            if not valid_ip_url:
-                parsed_url = urlparse(prometheus_url)
-                hostname = parsed_url.hostname
-                orch = OrchClient.instance()
-                inventory_hosts = [host.to_json() for host in orch.hosts.list()]
+            if '.ceph-dashboard' in prometheus_url:
+                prometheus_url = prometheus_url.replace('.ceph-dashboard', '')
+            parsed_url = urlparse(prometheus_url)
+            scheme = parsed_url.scheme
+            hostname = parsed_url.hostname
+            try:
+                # Check if the hostname is already an IP address
+                ipaddress.ip_address(hostname)
+                valid_ip_url = True
+            except ValueError:
+                valid_ip_url = False
+
+            orch = OrchClient.instance()
+            inventory_hosts = (
+                [host.to_json() for host in orch.hosts.list()]
+                if not valid_ip_url
+                else []
+            )
```

Comment:
This is poorly structured. From this check, 2 possible paths can be taken, but instead this unclear `valid_ip_url` is used multiple times to do alternative things...

BTW, this code breaks IPv6 again (IPv6 in URLs has to be enclosed in square brackets).

```suggestion
            try:
                # Check if the hostname is already an IP address
                ipaddress.ip_address(hostname)
                
                # IT's AN IPADDRESS!!
                node_ip = next(
                        (host for orch.hosts.list()
                        if host.hostname.startswith(hostname))
                        )
                        
                # BREAKS IPv6!!
                if mgmt_gw_enabled:
                    prometheus_url = f"{scheme}://{node_ip}"
                else:
                    prometheus_url = prometheus_url.replace(hostname, node_ip)
            except ValueError:
                # IT'S A HOSTNAME, DO THE STEPS ACCORDINLY
                if mgmt_gw_enabled:
                    prometheus_url = f"{scheme}://{hostname}"
                    
            return prometheus_url
```
That said, this code is hard to read and follow. It's not clear what it's doing and what for...